modules/extension/wmts/src/main/java/org/geotools/data/wmts/request/AbstractGetTileRequest.java

```diff
@@ -0,0 +1,401 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2017, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+
+package org.geotools.data.wmts.request;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.geotools.data.ows.HTTPResponse;
+import org.geotools.data.ows.Response;
+import org.geotools.data.wmts.client.WMTSTileService;
+import org.geotools.data.wmts.client.WMTSTileFactory;
+import org.geotools.data.wmts.model.WMTSCapabilities;
+import org.geotools.data.wmts.model.WMTSLayer;
+import org.geotools.data.wmts.model.TileMatrixLimits;
+import org.geotools.data.wmts.model.TileMatrixSet;
+import org.geotools.data.wmts.model.TileMatrixSetLink;
+import org.geotools.data.wmts.model.WMTSServiceType;
+import org.geotools.geometry.jts.ReferencedEnvelope;
+import org.geotools.ows.ServiceException;
+import org.geotools.referencing.CRS;
+import org.geotools.renderer.lite.RendererUtilities;
+import org.geotools.tile.Tile;
+
+import org.geotools.util.logging.Logging;
+import org.opengis.referencing.FactoryException;
+import org.opengis.referencing.crs.CoordinateReferenceSystem;
+import org.opengis.referencing.operation.TransformException;
+
+/**
+ * 
+ * (Based on existing work by rgould for WMS service)
+ * @author ian
+ * @author Emanuele Tajariol (etj at geo-solutions dot it)
+ */
+public abstract class AbstractGetTileRequest extends AbstractWMTSRequest
+        implements GetTileRequest {
+
+    /** MAXTILES */
+    private static final int MAXTILES = 256;
+
+    /** DPI */
+    private static final double DPI = 96.0;
```

Comment:
This is not the standard OGC DPI (which would be 91.something), where is it coming from?